### PR TITLE
Chat message 1

### PR DESCRIPTION
**Pull request by phong016688**

### CODE REVIEW

File: app/src/main/java/com/ccc/chatapp/data/model/Message.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.ccc.chatapp.data.model
+
+import com.google.firebase.Timestamp
+
+data class Message(var message: String, var time: Timestamp)
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/data/model/Message.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/data/model/Message.kt#L5"></span>The file Message.kt is not ending with a new line.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/repositories/MessageRepository.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                            emiter.onNext(it.toObject(Message::class.java))
+                        }
+                    }
+                    .addOnFailureListener {
+                        emiter.tryOnError(it)
+                    }
+            }
+        }
+    }
+}
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/repositories/MessageRepository.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/repositories/MessageRepository.kt#L51"></span>The file MessageRepository.kt is not ending with a new line.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.ccc.chatapp.screens.chat.chatfragment
+
+import com.ccc.chatapp.screens.BasePresenter
+
+interface ChatFragmentView {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt#L5"></span>The class or object ChatFragmentView is empty.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+}
+
+interface ChatFragmentPresenter : BasePresenter {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentContract.kt#L9"></span>The class or object ChatFragmentPresenter is empty.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.ccc.chatapp.screens.chat.chatfragment
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt#L1"></span>The file name ‘ChatFragmentPresenter.kt’ does not match the name of the single top-level declaration ‘ChatFragmentPresenterImpl’.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import com.ccc.chatapp.repositories.UserRepository
+import com.ccc.chatapp.utils.rx.SchedulerProvider
+
+class ChatFragmentPresenterImpl(
+    view: View?,
+    schedulerProvider: SchedulerProvider,
+    userRepository: UserRepository
+) : ChatFragmentPresenter {
+
+    override fun onStart() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt#L13"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    override fun onStop() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt#L16"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    override fun onDestroy() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/chatfragment/ChatFragmentPresenter.kt#L19"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/Adapter.kt [hunk 9]
```diff
@@ -0,0 +1,56 @@
+package com.ccc.chatapp.screens.chat.friendfragment
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/Adapter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/Adapter.kt#L1"></span>The file name ‘Adapter.kt’ does not match the name of the single top-level declaration ‘AdapterFriendList’.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragment.kt [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view: View = inflater.inflate(R.layout.fragment_account, container, false)
+        setUpRecyclerView(view)
+        mAdapter?.setItemClickListener(this)
+        return view
+    }
+
+    override fun goToChatFragment() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragment.kt#L61"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt [hunk 11]
```diff
@@ -0,0 +1,43 @@
+package com.ccc.chatapp.screens.chat.friendfragment
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt#L1"></span>The file name ‘ListFriendFragmentPresenter.kt’ does not match the name of the single top-level declaration ‘ListFriendFragmentPresenterImpl’.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+            .observeOn(schedulerProvider.ui())
+            .subscribe({
+                view?.updateListFriend(it)
+            }, {
+                Log.d(this::class.java.simpleName, it.message)
+            })
+        mCompositeDisposable.add(disposable)
+    }
+
+    override fun onStart() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt#L31"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    override fun onStop() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt#L34"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    override fun onDestroy() {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/truyenz/3C_ChatApp_Training/blob/dc182578086da9f4c0eb4b44ca76776c4077901e/app/src/main/java/com/ccc/chatapp/screens/chat/friendfragment/ListFriendFragmentPresenter.kt#L37"></span>This empty block of code can be removed.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/ccc/chatapp/data/model/Message.kt [hunk 15]
```diff
@@ -0,0 +1,5 @@
+package com.ccc.chatapp.data.model
+
+import com.google.firebase.Timestamp
+
+data class Message(var message: String = "", var time: Timestamp = Timestamp.now(), var iSend: Boolean = true)
```
Comment (onstonboy) [inline app/src/main/java/com/ccc/chatapp/data/model/Message.kt]: iSend -> isSent

File: app/src/main/java/com/ccc/chatapp/repositories/UserRepository.kt [hunk 16]
```diff
@@ -12,47 +15,65 @@ interface UserRepository {
...
     private var mFirestore: FirebaseFirestore = FirebaseFirestore.getInstance()
     private var mUser: User? = null
+    private var mToUser: User? = null
 
-    override fun isUserLogged(): Boolean {
-        return mAuth.currentUser != null
+    override fun getListFriend(): Observable<User> = Observable.create<DocumentSnapshot> { emiter ->
+        mFirestore.collection("user")
+            .get()
+            .addOnSuccessListener { snapshort ->
```
Comment (onstonboy) [inline app/src/main/java/com/ccc/chatapp/repositories/UserRepository.kt]: snapshort -> snapshot

File: app/src/main/java/com/ccc/chatapp/repositories/UserRepository.kt [hunk 17]
```diff
@@ -12,47 +15,65 @@ interface UserRepository {
...
     private var mUser: User? = null
+    private var mToUser: User? = null
 
-    override fun isUserLogged(): Boolean {
-        return mAuth.currentUser != null
+    override fun getListFriend(): Observable<User> = Observable.create<DocumentSnapshot> { emiter ->
+        mFirestore.collection("user")
+            .get()
+            .addOnSuccessListener { snapshot ->
+                for (i in snapshot.documents) {
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/repositories/UserRepository.kt]: rename i -> document

File: app/src/main/java/com/ccc/chatapp/screens/chat/ChatActivity.kt [hunk 18]
```diff
@@ -87,23 +103,31 @@ class ChatActivity : AppCompatActivity(), ChatView {
...
+                navigationView.selectedItemId = R.id.navigation_messenger
+            }
+        }
     }
 
     companion object {
-        fun getInstance(context: Context): Intent {
-            return Intent(context, ChatActivity::class.java)
-        }
+        const val Fragment_Friend = 0
```
Comment (truyenz) [inline app/src/main/java/com/ccc/chatapp/screens/chat/ChatActivity.kt]: FRAGMENT_FRIEND

**Pull request merged**